### PR TITLE
Fixed Like, User, Post Relations

### PR DESCRIPTION
**Pull request by PeterDMadrid**
**OVERVIEW**

- FIxed relationships between the like, post, user.
- Reverted to the old Like button (Removed Ajax)

**NEW FEATURE**
- 
image

### CODE REVIEW

File: src/app/Models/Post.php [hunk 1]
```diff
@@ -14,7 +14,11 @@ class Post extends Model
     use HasFactory;
     use SoftDeletes;
     
-    protected $fillable = ['content', 'user_id', 'childPost_id'];
+    protected $fillable = [
+        'content', 
+        'user_id', 
+        'childPost_id'
```
Comment (mikco-hipe) [inline src/app/Models/Post.php]: use snake case for 'childPost_id'

File: src/resources/views/pages/dashboard/post-footer.blade.php [hunk 2]
```diff
@@ -1,113 +1,124 @@
...
+        </div>
+    </div>
+    {{ $post->comments()->count() }} comments
+    {{ $post->comments()->count() }} shares
+</div>
+<div class="py-1">
+    <div class="flex justify-evenly m-0 relative border-t border-b border-gray-400 py-2">
+        @include('pages.dashboard.like-button')
+
+        <!-- comment -->
```
Comment (mikco-hipe) [inline src/resources/views/pages/dashboard/post-footer.blade.php]: remove comment

File: src/resources/views/pages/dashboard/post-footer.blade.php [hunk 3]
```diff
@@ -1,113 +1,124 @@
...
+                                                    'years',
+                                                ];
+                                                $replace = ['h', 'd', 's', 'm', 'w', 'wk', 'mn', 'y'];
+                                                $formattedTimeString = str_replace($search, $replace, $timeString);
+                                                $stringspace = str_replace(
+                                                    ' ',
+                                                    '',
+                                                    str_replace('ago', '', $formattedTimeString),
+                                                );
+                                            @endphp
```
Comment (mikco-hipe) [inline src/resources/views/pages/dashboard/post-footer.blade.php]: This feels like it should be from the backend

File: src/resources/views/pages/dashboard/post-footer.blade.php [hunk 4]
```diff
@@ -1,113 +1,124 @@
...
-                    </div>
-                </div>
+            @endforeach
+        </ul>
+    </div>
+    <div id="comment{{ $post->id }}"
+        class="d-dashboard__border-gray mb-2 flex p-2 items-center w-full hidden transition-all duration-300">
+        <div class="flex flex-col h-max-64 w-full">
+            <div>
+                <!-- Add Comment Form -->
```
Comment (mikco-hipe) [inline src/resources/views/pages/dashboard/post-footer.blade.php]: remove comment

File: src/resources/views/pages/dashboard/post-footer.blade.php [hunk 5]
```diff
@@ -1,113 +1,124 @@
...
+                                                    onclick="editComment({{ $comment->id }})">
+                                                    Edit
+                                                </span>
                                             @endif
-                                        </div>
-                                        <!-- Edit Comment Form -->
-                                        @include('pages.forms.edit-comment-form')
+                                        @endif
                                     </div>
+                                    <!-- Edit Comment Form -->
```
Comment (mikco-hipe) [inline src/resources/views/pages/dashboard/post-footer.blade.php]: remove comment

File: src/resources/views/pages/dashboard/post-footer.blade.php [hunk 6]
```diff
@@ -1,113 +1,124 @@
...
+    <div id="comment{{ $post->id }}"
+        class="d-dashboard__border-gray mb-2 flex p-2 items-center w-full hidden transition-all duration-300">
+        <div class="flex flex-col h-max-64 w-full">
+            <div>
+                <!-- Add Comment Form -->
+                @include('pages.forms.add-comment-form')
+            </div>
+        </div>
+    </div>
+
```
Comment (mikco-hipe) [inline src/resources/views/pages/dashboard/post-footer.blade.php]: new line

File: src/resources/views/pages/dashboard/post-content.blade.php [hunk 7]
```diff
@@ -1,108 +1,115 @@
...
+        </div>
     </div>
-</div>
 
-<!-- Modal Structure -->
-<div id="postModal" class="modal">
-    <div class="modal-content">
-        <span class="close" onclick="closeModal()">&times;</span>
-        <div id="modalPostContent"></div>
+    <!-- Modal Structure -->
```
Comment (mikco-hipe) [inline src/resources/views/pages/dashboard/post-content.blade.php]: remove comment

**Review by mikco-hipe — COMMENTED** [review]
Please check!

File: src/resources/views/pages/dashboard/post-footer.blade.php (hunk 3 above)
Comment (PeterDMadrid) [reply src/resources/views/pages/dashboard/post-footer.blade.php]: I inserted it on the post service. Now can be used in different view file without using @php @endphp

File: src/resources/views/pages/dashboard/post-footer.blade.php [hunk 8]
```diff
@@ -1,113 +1,99 @@
...
+    </div>
+    <div id="comment{{ $post->id }}"
+        class="d-dashboard__border-gray mb-2 flex p-2 items-center w-full hidden transition-all duration-300">
+        <div class="flex flex-col h-max-64 w-full">
+            <div>
+                @include('pages.forms.add-comment-form')
+            </div>
+        </div>
+    </div>
+
```
Comment (mikco-hipe) [inline src/resources/views/pages/dashboard/post-footer.blade.php]: I think theres an extra spaces to this line, please remove it.

Screenshot 2024-06-08 at 4 05 31 PM

File: src/app/Services/PostService.php [hunk 9]
```diff
@@ -77,15 +107,23 @@ public function editPost(Post $post, PostRequest $request): bool
...
 
     /**
      * Unlike Post
      */
     public function unlike(Post $post): void
     {
-        auth()->user()->likes()->detach($post);
+        $user = auth()->user();
+
+        $user->likes()->where('post_id', $post->id)->delete();
```
Comment (mikco-hipe) [inline src/app/Services/PostService.php]:
```suggestion
        auth()->user()->likes()->where('post_id', $post->id)->delete();
```
try this

**Review by mikco-hipe — COMMENTED** [review]
Please check!

File: src/app/Models/User.php [hunk 10]
```diff
@@ -57,6 +57,9 @@ public function profile(): HasOne
         return $this->hasOne(Profile::class, 'user_id');
     }
 
+    /**
+     * Send Email Verification
+     */
     public function sendEmailVerificationNotification()
```
Comment (hipe01mark) [inline src/app/Models/User.php]: return type

File: src/app/Models/Post.php [hunk 11]
```diff
@@ -32,9 +34,17 @@ public function photos(): HasMany
...
-    public function likes(): belongsToMany
+    public function likes(): hasMany
+    {
+        return $this->hasMany(Like::class, 'post_id');
+    }
+
+    /**
+     * Gets the user who liked the post
+     */
+    public function likedBy()
```
Comment (hipe01mark) [inline src/app/Models/Post.php]: return type

Comment (PeterDMadrid) [reply src/app/Models/Post.php]: Done

**Review by hipe01mark — APPROVED** [review]
LGTM 🙇

**Pull request merged**